src/incubator/TextField/CharCounter.tsx

```diff
@@ -6,22 +6,17 @@ import FieldContext from './FieldContext';
 
 export interface CharCounterProps {
   showCharCounter?: boolean;
-  maxLength?: number;
+  maxLength: number;
   charCounterStyle?: TextStyle;
 }
 
-export default ({showCharCounter, maxLength, charCounterStyle}: CharCounterProps) => {
+export default ({maxLength, charCounterStyle}: CharCounterProps) => {
   const {value} = useContext(FieldContext);
-
-  if (showCharCounter && !_.isUndefined(maxLength)) {
```

Comment:
!_.isUndefined(maxLength) should stay in the CharCounter component.

Reply:
But we declared maxLength is required.. so what's the point in checking it..

Reply:
when I do that.. 
I get a TS error in TextField code. 
Because I try to pass an optional value (maxLength) to a required prop (the maxLength in CharCounter)

Reply:
Make sense... Ok, let's just leave it as it was. If it is not meant to be standalone component then it doesn't really matter...